### PR TITLE
feat!: bumped node, react, major version

### PR DESCRIPTION
**Pull request by sashafklein**
Updates React and ReactDOM (to latest 18).
Updates recommended Node (to v20.8.0).
Bumps major version, to 3.0.0, because of incompatible dependencies. 

Also, moves React and ReactDOM into dependencies, which I think is the correct location, and out of Peer Dependencies.

### CODE REVIEW

**Review by nk1tz — APPROVED** [review]

**Comment by nk1tz** [comment]
Lets go!

**Review by nk1tz — APPROVED** [review]

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v16.14.0
+v20.8.0
```
Comment (robertshuford) [inline .nvmrc]: Lets use the current node `lts` which is `20.9.0`
```suggestion
v20.9.0
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@unchained/react-qr-reader",
-  "version": "2.3.0",
+  "version": "3.0.0",
```
Comment (robertshuford) [inline package.json]: Upstream going to `3.0.0-beta-1` makes this messy, and IMO setting ours to `2.4.0` is the "least worst" option.
```suggestion
  "version": "2.4.0",
```

**Comment by nk1tz** [comment]
@robertshuford I can't commit those changes here on github, but I approve 
✅

**Review by robertshuford — APPROVED** [review]

**Pull request merged**